lib/matplotlib/tests/test_backend_pdf.py

```diff
@@ -87,3 +87,22 @@ def test_multipage_keep_empty():
             pdf.savefig()
         assert os.path.exists(filename)
     os.remove(filename)
+
+@image_comparison(baseline_images=['do_not_combine_images'], remove_text=True)
+def test_do_not_combine_images():
+    #Test that figures can be saved without combining multiple images on a 
+    #single set of axes into a single image.  This test does not actually 
+    #verify that the images do not get combined, but it does ensure that mpl
+    #works properly when rcParams['pdf.combine_images'] = False.  The quickest
+    #way to verify that the images did not get combined is to open the vector
+    #graphics file in Adobe Illustrator or Inkscape and see if one image can 
+    #be moved relative to the other.
+    X, Y = np.meshgrid(np.arange(-5, 5, 1), np.arange(-5, 5, 1))
+    Z = np.sin(Y ** 2)
+
+    fig = plt.figure()
+    ax = fig.add_subplot(1, 1, 1)
+    ax.set_xlim(0, 3)
+    ax.imshow(Z, extent = [0, 1, 0, 1])
+    ax.imshow(Z[::-1], extent = [2, 3, 0, 1])
+    plt.rcParams['pdf.combine_images'] = False
```

Comment:
I think @mdboom is suggesting to grab the byte stream from the savefig call and use the sting tools to search for image commands.